packages/portal-contracts/test/erc20Gateway.ts

```diff
@@ -822,7 +861,8 @@ describe('ERC20 Gateway', async () => {
           blockInRoot,
           msgInBlock
         )
-      ).to.be.revertedWith('Message relay failed');
+      ).to.be.reverted;
```

Comment:
I have changed the way the FuelMessagePortal relays the messages so that it bubbles up the reason instead of giving a generic message. Also removed the use of the library it was using before because it was giving us no benefits over a normal call.

Some of these tests now fail with an underflow / overflow at the compiler level and I have not found a way to specify those with the current set of hardhat chai matchers. I 'll give it another look